src/main/webapp/app/exercises/programming/manage/programming-exercise.component.html

```diff
@@ -128,17 +142,33 @@
                             ></jhi-programming-exercise-instructor-status>
                         </div>
                         <div>
-                            <span *ngIf="programmingExercise.testRepositoryUrl"><a href="{{ programmingExercise.testRepositoryUrl }}" target="_blank">Test</a></span>
+                            <span *ngIf="programmingExercise.testRepositoryUrl"
+                                ><a *ngIf="!localVCEnabled; else downloadRepoLink" href="{{ programmingExercise.testRepositoryUrl }}" target="_blank">Test</a
+                                ><ng-template #downloadRepoLink
+                                    ><a [routerLink]="" (click)="downloadRepository(programmingExercise.id, 'TESTS')"><fa-icon [icon]="faDownload"></fa-icon> Test</a></ng-template
+                                ></span
+                            >
                         </div>
                     </td>
                     <td *jhiExtensionPoint="overrideRepositoryAndBuildPlan">
-                        <!-- TODO get the correct URL from the server instead of hardcoding Bamboo here -->
-                        <span *ngIf="programmingExercise.templateParticipation?.buildPlanId">
-                            <a target="_blank" rel="noreferrer" href="{{ programmingExercise.templateParticipation!.buildPlanUrl }}">Template</a>
-                        </span>
-                        <span *ngIf="programmingExercise.solutionParticipation && programmingExercise.solutionParticipation.buildPlanId">
-                            <a target="_blank" rel="noreferrer" href="{{ programmingExercise.solutionParticipation!.buildPlanUrl }}">Solution</a>
-                        </span>
+                        <ng-container *ngIf="programmingExercise.templateParticipation?.buildPlanId">
+                            <span *ngIf="!localVCEnabled; else plainBuildPlanId">
+                                <!-- TODO get the correct URL from the server instead of hardcoding Bamboo here -->
+                                <a target="_blank" rel="noreferrer" href="{{ programmingExercise.templateParticipation!.buildPlanUrl }}">Template</a>
+                            </span>
+                            <ng-template #plainBuildPlanId>
+                                <span>{{ programmingExercise.templateParticipation!.buildPlanId }}</span>
+                            </ng-template>
+                        </ng-container>
+                        <ng-container *ngIf="programmingExercise.solutionParticipation?.buildPlanId">
+                            <span *ngIf="!localVCEnabled; else plainBuildPlanId">
+                                <!-- TODO get the correct URL from the server instead of hardcoding Bamboo here -->
```

Comment:
Is this comment valid (and the one above)? Doesn't look like anything is hardcoded here.

Reply:
Doesn't look like it, seems to be something that was already fixed.